.gitlab/functional_test/regression_detector.yml

```diff
@@ -41,11 +52,15 @@ single-machine-performance-regression_detector:
     # Download smp binary and prepare it for use
     - aws --profile single-machine-performance s3 cp s3://smp-cli-releases/v${SMP_VERSION}/x86_64-unknown-linux-musl/smp smp
     - chmod +x smp
-    # Submit job, using the SHA of the last successful main commit as baseline.
-    # This will have been built in a separate pipeline run. The comparison will
-    # have been built previously in this pipeline run.
-    - aws --profile ${AWS_NAMED_PROFILE} s3 cp --only-show-errors s3://${SMP_AGENT_TEAM_ID}-smp-artifacts/information/latest_main_sha .latest_main_sha
-    - BASELINE_SHA=$(cat .latest_main_sha)
+    - BASELINE_SHA="${SMP_MERGE_BASE}"  # NOTE(geoffrey.oxberry@datadoghq.com): cat the echoed file once computing merge commit is separate job
+    - echo "Computing baseline..."
+    - echo "Checking if image exists for commit ${BASELINE_SHA}..."
+    - while [[ ! $(aws ecr describe-images --profile single-machine-performance --registry-id "${SMP_ACCOUNT_ID}" --repository-name "${SMP_AGENT_TEAM_ID}-agent" --image-ids imageTag="${BASELINE_SHA}-7-amd64") ]]; do echo "No image exists for ${BASELINE_SHA} - checking predecessor of ${BASELINE_SHA} next"; BASELINE_SHA=$(git rev-parse ${BASELINE_SHA}^); echo "Checking if image exists for commit ${BASELINE_SHA}..."; done
```

Comment:
@bkabrda I've added a loop that will iterate backwards over commits on `main` until it finds a commit with an image in SMP's ECR, starting with the merge base commit. Adding this loop should make the regression detector job match @KSerrania's "option 2" proposal for the regression detector baseline.